addons/pos_loyalty/static/src/overrides/components/product_screen/order_summary/order_summary.js

```diff
@@ -0,0 +1,104 @@
+/** @odoo-module **/
+
+import { _t } from "@web/core/l10n/translation";
+import { OrderSummary } from "@point_of_sale/app/screens/product_screen/order_summary/order_summary";
+import { patch } from "@web/core/utils/patch";
+import { ask } from "@point_of_sale/app/store/make_awaitable_dialog";
+import { useService } from "@web/core/utils/hooks";
+
+patch(OrderSummary.prototype, {
+    setup() {
+        super.setup(...arguments);
+        this.notification = useService("pos_notification");
+    },
+    async updateSelectedOrderline({ buffer, key }) {
+        const selectedLine = this.currentOrder.get_selected_orderline();
+        if (key === "-") {
+            if (selectedLine && selectedLine.eWalletGiftCardProgram) {
+                // Do not allow negative quantity or price in a gift card or ewallet orderline.
+                // Refunding gift card or ewallet is not supported.
+                this.notification.add(
+                    _t("You cannot set negative quantity or price to gift card or ewallet."),
+                    4000
+                );
+                return;
+            }
+        }
+        if (
+            selectedLine &&
+            selectedLine.is_reward_line &&
+            !selectedLine.manual_reward &&
+            (key === "Backspace" || key === "Delete")
+        ) {
+            const reward = this.pos.models["loyalty.reward"].get(selectedLine.reward_id);
+            const confirmed = await ask(this.dialog, {
```

Comment:
double check if `dialog` service is already 'used' in this component.

Reply:
in pos